### PR TITLE
[SPARK-43042] [SS] [Connect] Add table() API support for DataStreamReader

### PR DESCRIPTION
**Pull request by WweiL**
### What changes were proposed in this pull request?

Add the table() method support in DataStreamReader in Spark Connect.

### Why are the changes needed?

Continuation of building SS Connect

### Does this PR introduce _any_ user-facing change?

Yes now the table() API is available.

### How was this patch tested?

Unit test

### CODE REVIEW

**Comment by WweiL** [comment]
@rangadi @pengzhon-db @HyukjinKwon PTAL, thank you!

**Review by rangadi — APPROVED** [review]
LGTM

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 1]
```diff
@@ -874,6 +874,14 @@ class SparkConnectPlanner(val session: SparkSession) {
             s"Predicates are not supported for ${rel.getDataSource.getFormat} data sources.")
         }
 
+      case proto.Read.ReadTypeCase.NAMED_TABLE if rel.getIsStreaming =>
+        val multipartIdentifier =
+          CatalystSqlParser.parseMultipartIdentifier(rel.getNamedTable.getUnparsedIdentifier)
+        UnresolvedRelation(
+          multipartIdentifier,
+          new CaseInsensitiveStringMap(rel.getNamedTable.getOptionsMap),
+          isStreaming = true)
```
Comment (amaliujia) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:883]: hmm shouldn't this just be `isStreaming = rel.getIsStreaming`?

Comment (WweiL) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala]: Ah right that makes sense. Thank you!

**Comment by amaliujia** [comment]
Why do you need the change in `dev/tox.ini`?

**Comment by WweiL** [comment]
> Why do you need the change in `dev/tox.ini`?

@amaliujia Ah thanks for pointing it out. I'll revert it.

The change and reason is in this PR: https://github.com/apache/spark/pull/40801

**Review by amaliujia — APPROVED** [review]
LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**